util.js

```diff
@@ -61,7 +61,7 @@ function urlTemplate (opts) {
 }
 
 function getHostMirrorUrl (opts) {
-  var propName = 'npm_config_' + opts.pkg.name + '_binary_host'
+  var propName = 'npm_config_' + opts.pkg.name.replace(/-/g, '_') + '_binary_host'
```

Comment:
```suggestion
  var propName = 'npm_config_' + (opts.pkg.name || '').replace(/-/g, '_') + '_binary_host'
```

Unrelated tests are failing because they use an incomplete mock package, this is a workaround for that.